File: M3 Intermediate DSA/src/L8_Arrays_2D_Matrices/Assignment1.java
package L8_Arrays_2D_Matrices;

public class Assignment1 {

  public int[] solve(int[][] A) {
    int N = A.length;
    int M = A[0].length;
    int[] ans = new int[M];

    for (int col = 0; col < M;col++) {
      int sum = 0;
      for (int row = 0; row < N; row++) {
        sum += A[row][col];
      }
      ans[col] = sum;
    }

    return ans;
  }

}

File: M3_Intermediate_DSA/L6_2D_Matrices/assignment1.go
package main

func SolveA1(A [][]int) []int {
	N := len(A)
	M := len(A[0])
	ans := make([]int, M)

	for col := 0; col < M; col++ {
		var sum int
		for row := 0; row < N; row++ {
			sum += A[row][col]
		}
		ans[col] = sum
	}

	return ans
}
